architecture/2025-04-compute-platform-extensibility.md

```diff
@@ -0,0 +1,505 @@
+
+```bicep
+extension radius
+@description('Specifies the environment for resources.')
+param environment string
+
+@description('Specifies the application for resources.')
+param application string
+
+resource redis 'Applications.Datastores/redisCaches@2023-10-01-preview'= {
+  name: 'myredis'
+  properties: {
+    environment: environment
+    application: application
+  }
+}
+```
+
+#### Output
+
+The above example will deploy on either Kubernetes or ACI.
+
+## Design
+
+### High Level Design
+
+The integration with ACI will be achieved by extending the existing Radius control plane, which will continue to be hosted in Kubernetes, to support ACI as a target deployment environment. The following summarizes the key changes and enhancements that will be made to the Radius codebase to support ACI deployments:
+
+1. Environment Definition and Setup: The Radius control plane will be updated to recognize and process environment definition files (env.bicep) that specify ACI as the target compute platform. This will involve adding support for ACI-specific configurations and parameters in the environment setup process.
+1. Application Definition and Deployment: The application definition files (app.bicep) will be extended to include ACI-specific properties and configurations. The Radius deployment engine will be enhanced to interpret these properties and deploy applications to ACI environments. This includes creating and managing ACI container groups, configuring networking, and handling ACI-specific resource settings.
+1. Container Definitions: The container definitions within the application definition files will allow users to deploy their containers to both Kubernetes and ACI without significant changes.
+1. Extension Properties, aka "punch-through": The Radius codebase will be enhanced to support extension properties in the application definition files. These properties will allow users to apply ACI-specific configurations to their containers, such as custom resource settings, networking configurations, and other ACI-specific parameters.
+1. Error Handling and Logging: The error handling and logging mechanisms within the Radius codebase will be updated to account for ACI-specific scenarios. This includes handling errors related to ACI resource creation, configuration, and deployment, as well as providing detailed logs for troubleshooting and diagnostics.
+1. Gateways: Radius will provision the necessary Azure resources to establish gateways for routing internet traffic to services hosted in ACI. This includes defining and creating gateways and secret stores for use in ACI containers.
+1. TBD: Secret Management: Radius will provision an Azure KeyVault resource on behalf of the user to store and manage secrets. These secrets can then be referenced in the application definition files to inject secret values as environment variables in ACI containers or to authenticate into services.
```

Comment:
Is this different from currently supported `Applications.Core/SecretStore` resource? If not, then this is really enabling another type of secret manager for the Radius secret store resource, rest of the functionality remains the same as it is today regardless of where the app gets deployed to.

Reply:
Yes, we would implement an Azure KeyVault deployment for the `Applications.Core/SecretStore`. Doing so is not in scope for the first release.